src/main/java/io/cdap/plugin/snowflake/common/OAuthUtil.java

```diff
@@ -50,9 +55,19 @@ public static String getAccessTokenByRefreshToken(CloseableHttpClient httpclient
       httppost.setHeader("Content-type", "application/x-www-form-urlencoded");
 
       // set grant type and refresh_token. It should be in body not url!
-      StringEntity entity = new StringEntity(String.format("refresh_token=%s&grant_type=refresh_token",
-                                                           URLEncoder.encode(config.getRefreshToken(), "UTF-8")));
-      httppost.setEntity(entity);
+      try {
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(config.getRefreshToken());
+        StringEntity entity = new StringEntity(String.format("refresh_token=%s&grant_type=refresh_token",
+          URLEncoder.encode(config.getRefreshToken(), "UTF-8")));
+        httppost.setEntity(entity);
+      } catch (NullPointerException e) {
+        String errorMessage = String.format("Error encoding URL due to missing Refresh Token. Failure reason is %s.",
+          e.getMessage());
+        String errorReason = "Error encoding URL due to missing Refresh Token.";
+        throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+          errorReason, errorMessage, ErrorType.SYSTEM, true, e);
```

Comment:
why is this `SYSTEM` error, if the refresh token is provided by cx then it should be `USER` error.